### PR TITLE
config: adds github workflow ci

### PR DESCRIPTION
**Pull request by rnveach**
Adds github workflow to take over for travis for now.

### CODE REVIEW

**Comment by rnveach** [comment]
https://github.com/checkstyle/eclipse-cs/actions/runs/4070627043/jobs/7011701526
````
[INFO] --- exec-maven-plugin:3.0.0:exec (default) @ net.sf.eclipsecs.parent ---
Error:  Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 128 (Exit value: 128)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
````

I am not sure what the failure means.

File: pom.xml [hunk 1]
```diff
@@ -111,28 +111,6 @@
...
-                        <goal>exec</goal>
-                      </goals>
-                    </execution>
-                </executions>
-                <configuration>
-                    <executable>git</executable>
-                    <arguments>
-                      <argument>diff</argument>
-                      <argument>HEAD~1</argument>
-                      <argument>HEAD</argument>
```
Comment (rnveach) [inline pom.xml:131]: It looks like this command isn't working in the action.

````
Run git diff HEAD~1 HEAD > show.patch
fatal: ambiguous argument 'HEAD~1': unknown revision or path not in the working tree.
````

Comment (rnveach) [reply pom.xml:131]: Issue seems to be fetch depth needs to be configured so we have an actual depth to check.
https://github.com/actions/checkout#usage
````
    # Number of commits to fetch. 0 indicates all history for all branches and tags.
    # Default: 1
    fetch-depth: ''
````

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  install:
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 50
```
Comment (rnveach) [inline .github/workflows/maven.yml:16]: Travis use to do all the commits in the PR. I don't know what github workflow will do so I made this a bigger number for now.

**Review by romani — APPROVED** [review]
Let's merge this, and move more jobs as able to do this

**Pull request merged**